services/app/src/components/views/quest/cardtemplates/decision/Actions.test.tsx

```diff
@@ -1,39 +1,212 @@
+          "rolls": [],
+          "selected": null
+        });
+      expect(actions[2].to).toEqual(jasmine.objectContaining({phase: 'PREPARE_DECISION'}));
+    });
   });
   describe('computeSuccesses', () => {
-    test.skip('works when zero rolls', () => { /* TODO */ });
-    test.skip('counts successes and ignores other rolls', () => { /* TODO */ });
-    test.skip('respects difficulty', () => { /* TODO */ });
+    test('works when zero rolls', () => {
+      expect(computeSuccesses([], {difficulty: 'medium'})).toEqual(0);
+    });
+    test('counts successes and ignores other rolls', () => {
+      expect(computeSuccesses([5, 20, 10, 5], {difficulty: 'medium'})).toEqual(1);
+    });
+    test('respects difficulty', () => {
+      expect(computeSuccesses([16, 15, 10, 14], {difficulty: 'hard'})).toEqual(0);
+    });
   });
   describe('computeOutcome', () => {
-    test.skip('computes success', () => { /* TODO */ });
-    test.skip('computes failure', () => { /* TODO */ });
-    test.skip('computes interrupted', () => { /* TODO */ });
-    test.skip('computes retry', () => { /* TODO */ });
-    test.skip('returns null when no selected decision', () => { /* TODO */ });
+    const selected = {difficulty: 'medium', requiredSuccesses: 5};
+
+    test('computes success', () => {
+      expect(computeOutcome([20, 20, 20, 20, 20], selected, TEST_SETTINGS, TEST_RP)).toEqual(Outcome.success);
+    });
+    test('computes failure', () => {
+      expect(computeOutcome([1], selected, TEST_SETTINGS, TEST_RP)).toEqual(Outcome.failure);
+    });
+    test('computes interrupted', () => {
+      expect(computeOutcome([20, 20, 20, 20, 10], selected, TEST_SETTINGS, TEST_RP)).toEqual(Outcome.interrupted);
+    });
+    test('computes retry', () => {
+      expect(computeOutcome([20, 20, 20, 20], selected, TEST_SETTINGS, TEST_RP)).toEqual(Outcome.retry);
+    });
+    test('returns null when no rolls', () => {
+      expect(computeOutcome([], selected, TEST_SETTINGS, TEST_RP)).toEqual(null);
+    });
   });
   describe('generateLeveledChecks', () => {
-    test.skip('returns 3 semi-unique, generated checks', () => { /* TODO */ });
-    test.skip('scales num successes with num adventurers', () => { /* TODO */ });
+    test('returns 3 semi-unique, generated checks', () => {
+      expect(generateLeveledChecks(5, seedrandom.alea('1234'))).toEqual([
+        {difficulty: 'hard', persona: 'light', requiredSuccesses: 1, skill: 'athletics'},
+        {difficulty: 'medium', persona: 'dark', requiredSuccesses: 3, skill: 'athletics'},
+        {difficulty: 'easy', persona: 'light', requiredSuccesses: 1, skill: 'knowledge'},
+      ]);
+    });
+    test('scales num successes with num adventurers', () => {
+      expect(Math.max(...generateLeveledChecks(1, seedrandom.alea('1234')).map((c) => c.requiredSuccesses))).toEqual(1);
+      expect(Math.max(...generateLeveledChecks(6, seedrandom.alea('1234')).map((c) => c.requiredSuccesses))).toBeGreaterThan(3);
```

Comment:
Wait, so with 6 players.... at least 4 players would have to roll and succeed? That seems a bit much, I would expect 3 at most, both statistically and so that it doesn't slow down the game too much with all the extra rolling

Reply:
Capped at 3.